sklearn/linear_model/base.py

```diff
@@ -112,6 +112,19 @@ def center_data(X, y, fit_intercept, normalize=False, copy=True,
     return X, y, X_mean, y_mean, X_std
 
 
+def rescale_data(X, y, sample_weight):
+    """Rescale data so as to support sample_weight"""
+    n_samples = X.shape[0]
+    sample_weight = sample_weight * np.ones(n_samples)
```

Comment:
Do we need to support a scalar `sample_weight`? Does that really change the result or does it wash out?